### DISCUSSION TITLE
Only exit process with CTRL+C when on empty line

### DISCUSSION THREAD

**Discussion by domdorn** [sbt/sbt#7465] [Ideas]
I find myself accidently quiting sbt with CTRL+C when all I want is to clear the current line. 

E.g. I'm also using a different keyboard layout than english and just now have written
```
testOnlz com.acme.project.masterdata.persistence.eventsourced.scenariomanager.ScenarioManagerBehaviorsSpec
```

instead of 
```
testOnly com.acme.project.masterdata.persistence.eventsourced.scenariomanager.ScenarioManagerBehaviorsSpec
```

Instinctively I pressed CTRL+C to clear the current line (e.g. same behaviour as in any normal terminal or psql)
Instead of just clearing the current line, the sbt process aborted. 

I'd like to propose that the handler operating on CTRL+C should work like following:
```
if current_line is not empty
 clear current line / create a new line without executing what was written on current line
else 
 stop process / exit sbt
```

**Comment by eed3si9n** [comment]
Thanks for bringing this up. If it's not too difficult to implement it, we should support this use case.
